pwnlib/libcdb.py

```diff
@@ -126,7 +127,23 @@ def provider_local_system(hex_encoded_id, hash_type):
         return local_libc.data
     return None
 
-PROVIDERS = [provider_local_system, provider_libcdb, provider_libc_rip]
+# Offline search https://github.com/niklasb/libc-database for hash type
+def provider_local_database(hex_encoded_id, hash_type):
+    assert context.local_libcdb
+
+    localdb = Path(context.local_libcdb)
+    if not localdb.is_dir():
+        log.warn_once("%s does not exist, please download libc-database first.", str(localdb))
```

Comment:
Hm, can we somehow suppress this warning if you didn't change the `local_libcdb` option? This would print everytime now by default even when you're not trying to use a local libc-database.

Reply:
Or we can set the default value of `context.local_libcdb` to -1, and make the `provider_local_database` function return None directly, unless it is set to a string.

Reply:
Having a default location for the files is useful, so the easiest would be to check if the value is something else than the default.
You can check `if 'local_libcdb' in context._tls` to see if the user modified it in the code, but that won't catch changes set through the pwn.conf or `LOCAL_LIBCDB` argument, since they modify the `ContextType.defaults` directly. So easiest would be to put the default path in some constant and check if it's different.